File: Archiver/CS/clsATTACHMENTTYPE.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using System.Drawing;
using System.Linq;
using System.Diagnostics;
using System.Data;
using Microsoft.VisualBasic;
using MODI;
using System.Xml.Linq;
using System.Collections;
using System.Windows.Forms;
// End of VB project level imports

using System.Configuration;
using System.Data.SqlClient;
using System.IO;
using System.Data.Sql;


namespace EcmArchiveClcSetup
{
	public class clsATTACHMENTTYPE
	{
		
		//** DIM the selected table columns
		clsDatabase DB = new clsDatabase();
		clsDma DMA = new clsDma();
		
		clsUtility UTIL = new clsUtility();
		clsLogging LOG = new clsLogging();
		
		
		string AttachmentCode = "";
		string Description = "";
		string isZipFormat = "";
		
		
		//** Generate the SET methods
		public void setAttachmentcode(ref string val)
		{
			if (val.Length == 0)
			{
				MessageBox.Show("SET: Field \'Attachmentcode\' cannot be NULL.");
				return;
			}
			val = UTIL.RemoveSingleQuotes(val);
			AttachmentCode = val;
		}
		
		public void setDescription(ref string val)
		{
			val = UTIL.RemoveSingleQuotes(val);
			Description = val;
		}
		
		public void setIszipformat(ref string val)
		{
			if (val.Length == 0)
			{
				val = "null";
			}
			val = UTIL.RemoveSingleQuotes(val);
			isZipFormat = val;
		}
		
		
		
		//** Generate the GET methods
		public string getAttachmentcode()
		{
			if (AttachmentCode.Length == 0)
			{
				MessageBox.Show("GET: Field \'Attachmentcode\' cannot be NULL.");
				return "";
			}
			return UTIL.RemoveSingleQuotes(AttachmentCode);
		}
		
		public string getDescription()
		{
			return UTIL.RemoveSingleQuotes(Description);
		}
		
		public string getIszipformat()
		{
			if (isZipFormat.Length == 0)
			{
				isZipFormat = "null";
			}
			return isZipFormat;
		}
		
		
		
		//** Generate the Required Fields Validation method
		public bool ValidateReqData()
		{
			if (AttachmentCode.Length == 0)
			{
				return false;
			}
			return true;
		}
		
		
		//** Generate the Validation method
		public bool ValidateData()
		{
			if (AttachmentCode.Length == 0)
			{
				return false;
			}
			return true;
		}
		
		
		//** Generate the INSERT method
		public bool Insert()
		{
			bool b = false;
			string s = "";
			
			if (AttachmentCode.Trim().Length > 10)
			{
				AttachmentCode = "UKN";
				Description = "Unknown Attachment Type";
			}
			
			if (isZipFormat.Length == 0)
			{
				isZipFormat = "0";
			}
			
			s = s + " INSERT INTO AttachmentType(";
			s = s + "AttachmentCode,";
			s = s + "Description,";
			s = s + "isZipFormat) values (";
			s = s + "\'" + AttachmentCode + "\'" + ",";
			s = s + "\'" + Description + "\'" + ",";
			s = s + isZipFormat + ")";
			return DB.ExecuteSqlNewConn(s, false);
			
		}
		
		
		//** Generate the UPDATE method
		public bool Update(string WhereClause)
		{
			bool b = false;
			string s = "";
			
			if (WhereClause.Length == 0)
			{
				return false;
			}
			
			s = s + " update AttachmentType set ";
			//s = s + "AttachmentCode = '" + getAttachmentcode() + "'" + ", "
			s = s + "Description = \'" + getDescription() + "\'" + ", ";
			s = s + "isZipFormat = " + getIszipformat();
			WhereClause = (string) (" " + WhereClause);
			s = s + WhereClause;
			return DB.ExecuteSqlNewConn(s, false);
		}
		
		
		//** Generate the SELECT method
		public SqlDataReader SelectRecs()
		{
			bool b = false;
			string s = "";
			SqlDataReader rsData;
			s = s + " SELECT ";
			s = s + "AttachmentCode,";
			s = s + "Description,";
			s = s + "isZipFormat ";
			s = s + " FROM AttachmentType";
			
			string CS = DB.getGateWayConnStr(modGlobals.gGateWayID);
			SqlConnection CONN = new SqlConnection(CS);
			CONN.Open();
			SqlCommand command = new SqlCommand(s, CONN);
			rsData = command.ExecuteReader();
			return rsData;
		}
		
		
		//** Generate the Select One Row method
		public SqlDataReader SelectOne(string WhereClause)
		{
			bool b = false;
			string s = "";
			SqlDataReader rsData;
			s = s + " SELECT ";
			s = s + "AttachmentCode,";
			s = s + "Description,";
			s = s + "isZipFormat ";
			s = s + " FROM AttachmentType";
			s = s + WhereClause;
			
			string CS = DB.getGateWayConnStr(modGlobals.gGateWayID);
			SqlConnection CONN = new SqlConnection(CS);
			CONN.Open();
			SqlCommand command = new SqlCommand(s, CONN);
			rsData = command.ExecuteReader();
			return rsData;
		}
		
		
		//** Generate the DELETE method
		public bool Delete(string WhereClause)
		{
			bool b = false;
			string s = "";
			
			if (WhereClause.Length == 0)
			{
				return false;
			}
			
			WhereClause = (string) (" " + WhereClause);
			
			s = " Delete from AttachmentType";
			s = s + WhereClause;
			
			b = DB.ExecuteSqlNewConn(s, false);
			return b;
			
		}
		
		
		//** Generate the Zeroize Table method
		public bool Zeroize()
		{
			bool b = false;
			string s = "";
			
			s = s + " Delete from AttachmentType";
			
			b = DB.ExecuteSqlNewConn(s, false);
			return b;
			
		}
		
		
		//** Generate Index Queries
		public int cnt_PK29(string AttachmentCode)
		{
			
			int B = 0;
			string TBL = "AttachmentType";
			string WC = "Where AttachmentCode = \'" + AttachmentCode + "\'";
			
			B = DB.iGetRowCount(TBL, WC);
			
			return B;
		} //** cnt_PK29
		
		//** Generate Index ROW Queries
		public SqlDataReader getRow_PK29(string AttachmentCode)
		{
			
			SqlDataReader rsData = null;
			string TBL = "AttachmentType";
			string WC = "Where AttachmentCode = \'" + AttachmentCode + "\'";
			
			rsData = DB.GetRowByKey(TBL, WC);
			
			if (rsData.HasRows)
			{
				return rsData;
			}
			else
			{
				return null;
			}
		} //** getRow_PK29
		
		/// Build Index Where Caluses
		///
		public string wc_PK29(string AttachmentCode)
		{
			
			string WC = "Where AttachmentCode = \'" + AttachmentCode + "\'";
			
			return WC;
		} //** wc_PK29
		
		//** Generate the SET methods
		
	}
	
}

File: SearchSVC/clsATTACHMENTTYPE.vb
Imports Microsoft.VisualBasic
Imports System.Configuration
Imports System
Imports System.Data.SqlClient
Imports System.Collections
Imports System.IO
Imports System.Data.Sql
 
Public Class clsATTACHMENTTYPE

    '** DIM the selected table columns 
    Dim DB As New clsDatabaseSVR

    Dim DMA As New clsDmaSVR

    Dim UTIL As New clsUtilitySVR
    Dim LOG As New clsLogging


    Dim AttachmentCode As String = ""
    Dim Description As String = ""
    Dim isZipFormat As String = ""

    Dim SecureID As Integer = -1

    Sub New(ByVal iSecureID As Integer)
        SecureID = iSecureID
    End Sub


    '** Generate the SET methods 
    Public Sub setAttachmentcode(ByRef val as string)
        If Len(val) = 0 Then
            MsgBox("SET: Field 'Attachmentcode' cannot be NULL.")
            Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        AttachmentCode = val
    End Sub

    Public Sub setDescription(ByRef val as string)
        val = UTIL.RemoveSingleQuotes(val)
        Description = val
    End Sub

    Public Sub setIszipformat(ByRef val as string)
        If Len(val) = 0 Then
            val = "null"
        End If
        val = UTIL.RemoveSingleQuotes(val)
        isZipFormat = val
    End Sub



    '** Generate the GET methods 
    Public Function getAttachmentcode() As String
        If Len(AttachmentCode) = 0 Then
            MsgBox("GET: Field 'Attachmentcode' cannot be NULL.")
            Return ""
        End If
        Return UTIL.RemoveSingleQuotes(AttachmentCode)
    End Function

    Public Function getDescription() As String
        Return UTIL.RemoveSingleQuotes(Description)
    End Function

    Public Function getIszipformat() As String
        If Len(isZipFormat) = 0 Then
            isZipFormat = "null"
        End If
        Return isZipFormat
    End Function



    '** Generate the Required Fields Validation AS string 
    Public Function ValidateReqData() As Boolean
        If AttachmentCode.Length = 0 Then Return false
        Return True
    End Function


    '** Generate the Validation AS string 
    Public Function ValidateData() As Boolean
        If AttachmentCode.Length = 0 Then Return false
        Return True
    End Function


    '** Generate the INSERT AS string 
    Public Function Insert() As Boolean
        Dim b As Boolean = false
        Dim s As String = ""

        If AttachmentCode.Trim.Length > 10 Then
            AttachmentCode = "UKN"
            Description = "Unknown Attachment Type"
        End If

        If isZipFormat.Length = 0 Then
            isZipFormat = "0"
        End If

        s = s + " INSERT INTO AttachmentType("
        s = s + "AttachmentCode,"
        s = s + "Description,"
        s = s + "isZipFormat) values ("
        s = s + "'" + AttachmentCode + "'" + ","
        s = s + "'" + Description + "'" + ","
        s = s + isZipFormat + ")"
        Return DB.DBExecuteSql(SecureID, S, False)

    End Function


    '** Generate the UPDATE AS string 
    Public Function Update(ByVal WhereClause as string) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        If Len(WhereClause) = 0 Then Return False

        s = s + " update AttachmentType set "
        's = s + "AttachmentCode = '" + getAttachmentcode() + "'" + ", "
        s = s + "Description = '" + getDescription() + "'" + ", "
        s = s + "isZipFormat = " + getIszipformat()
        WhereClause = " " + WhereClause
        s = s + WhereClause
        Return DB.DBExecuteSql(SecureID, S, False)
    End Function


    '** Generate the SELECT AS string 
    Public Function SelectRecs() As SqlDataReader
        Dim b As Boolean = False
        Dim s As String = ""
        Dim rsData As SqlDataReader
        s = s + " SELECT "
        s = s + "AttachmentCode,"
        s = s + "Description,"
        s = s + "isZipFormat "
        s = s + " FROM AttachmentType"

        Dim CS$ = DBgetConnStr() : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
        Return rsData
    End Function


    '** Generate the Select One Row AS string 
    Public Function SelectOne(ByVal WhereClause as string) As SqlDataReader
        Dim b As Boolean = False
        Dim s As String = ""
        Dim rsData As SqlDataReader
        s = s + " SELECT "
        s = s + "AttachmentCode,"
        s = s + "Description,"
        s = s + "isZipFormat "
        s = s + " FROM AttachmentType"
        s = s + WhereClause

        Dim CS$ = DBgetConnStr() : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
        Return rsData
    End Function


    '** Generate the DELETE AS string 
    Public Function Delete(ByVal WhereClause as string) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        If Len(WhereClause) = 0 Then Return False

        WhereClause = " " + WhereClause

        s = " Delete from AttachmentType"
        s = s + WhereClause

        b = DB.DBExecuteSql(SecureID, S, False)
        Return b

    End Function


    '** Generate the Zeroize Table AS string 
    Public Function Zeroize() As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        s = s + " Delete from AttachmentType"

        b = DB.DBExecuteSql(SecureID, S, False)
        Return b

    End Function


    '** Generate Index Queries 
    Public Function cnt_PK29(ByVal AttachmentCode As String) As Integer

        Dim B As Integer = 0
        Dim TBL$ = "AttachmentType"
        Dim WC$ = "Where AttachmentCode = '" + AttachmentCode + "'"

        B = DB.iGetRowCount(TBL$, WC)

        Return B
    End Function     '** cnt_PK29

    '** Generate Index ROW Queries 
    Public Function getRow_PK29(ByVal AttachmentCode As String) As SqlDataReader

        Dim rsData As SqlDataReader = Nothing
        Dim TBL$ = "AttachmentType"
        Dim WC$ = "Where AttachmentCode = '" + AttachmentCode + "'"

        rsData = DB.GetRowByKey(SecureID, TBL$, WC)

        If rsData.HasRows Then
            Return rsData
        Else
            Return Nothing
        End If
    End Function     '** getRow_PK29

    ''' Build Index Where Caluses 
    '''
    Public Function wc_PK29(ByVal AttachmentCode As String) As String

        Dim WC$ = "Where AttachmentCode = '" + AttachmentCode + "'"

        Return WC
    End Function     '** wc_PK29

    '** Generate the SET methods 

End Class
